src/types/list.rs

```diff
@@ -720,4 +773,124 @@ mod tests {
             assert!(list.index(42i32).is_err());
         });
     }
+
+    use std::ops::Range;
+
+    struct FaultyIter(Range<usize>, usize);
```

Comment:
Might be worth adding a doc and/or using named fields to make it clearer what's "faulty" about this.